acra/guides/integrating-acra-translator-into-new-infrastructure/_index.md

```diff
@@ -4,55 +4,123 @@ bookCollapseSection: true
 
+## Key generation
+
+You will need some keys in order to launch AcraServer, so let's do it first.
+
+1. [Generate a master key]({{< ref "acra/security-controls/key-management/operations/generation.md#11-generating-acra-master-key">}})
+2. [Generate encryption keys]({{< ref "acra/security-controls/key-management/operations/generation.md#12-generating-transport-and-encryption-keys">}})
+
+The first one will be used to protect all the other keys,
+it should be base64-encoded and passed to Acra services in `ACRA_MASTER_KEY` environment variable.
+
+Like this: `ACRA_MASTER_KEY="$(cat /tmp/master_key | base64)" acra-server ...`
+
+The second key is responsible for data encryption.
+There are actually more kinds of keys, read more about that on
+[Acra keys inventory]({{< ref "acra/security-controls/key-management/inventory.md" >}}).
+
+It is also possible to store keys in a Redis database, see
+[Scalable KV storages]({{< ref "acra/configuring-maintaining/key-storing/kv-stores.md#redis" >}}).
+
+### Note about Client ID
+
+When generating a key, you will always have to bind it with a [Cliend ID]({{< ref "client_id.md" >}}) or Zone ID.
+AcraTranslator distinguishes requests by [Client ID]({{< ref "client_id.md" >}}) and uses different encryption keys for 
+different client ID.
 
 ## Architecture and DataFlow of AcraTranslator-based infrastructure
 
 ![](/files/data-flow/acra-archi-translator-writer.png)
 
 Acra design values simplicity as much as security.
 
-1. The application sends [AcraStructs](/acra/acra-in-depth/data-structures/acrastruct) or [AcraBlocks](/acra/acra-in-depth/data-structures/acrablock) via HTTP or gRPC API.
-2. AcraConnector sends that request to AcraTranslator using [Themis Secure Session](/themis/crypto-theory/cryptosystems/secure-session/) (socket protection protocol).
-3. AcraTranslator accepts the request and attempts to decrypt an [AcraStruct](/acra/acra-in-depth/data-structures/acrastruct) or [AcraBlock](/acra/acra-in-depth/data-structures/acrablock). If the decryption is successful, it sends the resulting plaintext in the response. If decryption fails, AcraTranslator sends out a decryption error.
-4. AcraTranslator returns the data to AcraConnector (via [Themis Secure Session](/themis/crypto-theory/cryptosystems/secure-session/)), which in turn returns it to the application.
+1. The application sends [AcraStructs] or [AcraBlocks] via HTTP or gRPC API.
+2. AcraConnector sends that request to AcraTranslator using [Themis Secure Session] (socket protection protocol).
+3. AcraTranslator accepts the request and attempts to decrypt an [AcraStruct] or [AcraBlock]. If the decryption is
+   successful, it sends the resulting plaintext in the response. If decryption fails, AcraTranslator sends out a
+   decryption error.
+4. AcraTranslator returns the data to AcraConnector (via [Themis Secure Session]), which in turn returns it to the application.
```

Comment:
are you sure that putting links to the bottom makes sense?

Reply:
for example, you've copy-pasted a piece of text from AT to AS, but forgot to copy-paste links because they are hidden in the bottom. I'd recommend keeping links inline every time for easier maintenance of the docs.

Reply:
about copy-paste, it is my fault. 
but for me it was hard to read markdown text where one line with two links takes 2 horizontal scrolls. plus they used several times in the text... This block of text is good example. 2 reps of "secure session", 2 reps of "acrastructs" and "acrablocks". and every link takes 1/3 of row length... Or need to split and move them to next line for the next easier reading.
but okay, if it is better only for me, I will back inline links. maybe you are right and it worse for maintaining